### PR TITLE
Views for managing and attempting text response questions

### PR DESCRIPTION
**Pull request by BenMQ**
Not too sure how the grading part can be done yet.

### CODE REVIEW

File: app/controllers/course/assessment/submissions_controller.rb [hunk 1]
```diff
@@ -84,9 +84,14 @@ def update_answers_params
...
   def update_answer_type_params
-    {}.tap do |result|
-      result[:option_ids] = [] # MRQ answers
+    scalar_params = [].tap do |result|
+      result.push(:answer_text) # Text response answer
     end
+    # Parameters that must be an array of permitted values
+    array_params = {}.tap do |result|
+      result[:option_ids] = [] # MRQ answer
+    end
```
Comment (BenMQ) [inline app/controllers/course/assessment/submissions_controller.rb]: this thing took me 2 days to figure out on and off...

Comment (lowjoel) [inline app/controllers/course/assessment/submissions_controller.rb]: nice fix.

File: spec/features/course/assessment/question/text_response_management_spec.rb [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        fill_in 'title', with: question_attributes[:title]
+        fill_in 'description', with: question_attributes[:description]
+        fill_in 'maximum_grade', with: question_attributes[:maximum_grade]
+        click_button 'submit'
+
+        expect(current_path).to eq(course_assessment_path(course, assessment))
+        expect(page).to have_content_tag_for(assessment.questions.first.specific)
+      end
+
+      scenario 'I can edit a question', js: true do
```
Comment (lowjoel) [inline spec/features/course/assessment/question/text_response_management_spec.rb]: Do you need js?

File: spec/features/course/assessment/question/text_response_management_spec.rb [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        fill_in 'maximum_grade', with: maximum_grade
+        click_button 'submit'
+
+        expect(current_path).to eq(course_assessment_path(course, assessment))
+        expect(question.reload.maximum_grade).to eq(maximum_grade)
+
+        visit edit_path
+
+        solutions.each_with_index do |solution, i|
+          click_link I18n.t('course.assessment.question.text_responses.form.add_solution')
```
Comment (lowjoel) [inline spec/features/course/assessment/question/text_response_management_spec.rb]: OK yes you do.

File: spec/features/course/assessment/question/text_response_management_spec.rb [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+          within all('.edit_question_text_response '\
+            'tr.question_text_response_solution')[i] do
+            fill_in find('textarea.text-response-solution')[:name], with: solution[:solution]
+            fill_in find('textarea.text-response-explanation')[:name],
+                    with: solution[:explanation]
+            solution_type = find('select.text-response-solution-type', visible: :all)
+            # Twitter Bootstrap hides <select> element and creates a div.
+            # The usual #select method is broken as it does not seem to work with hidden elements.
```
Comment (lowjoel) [inline spec/features/course/assessment/question/text_response_management_spec.rb]: Yeah this is a problem with the component I'm using for styling...

**Pull request merged**